mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearRegressionWithSGD.scala

```diff
@@ -47,6 +52,7 @@ class StreamingLinearRegressionWithSGD private[mllib] (
     private var numIterations: Int,
     private var miniBatchFraction: Double)
   extends StreamingLinearAlgorithm[LinearRegressionModel, LinearRegressionWithSGD]
+  with StreamingDecaySetter[StreamingLinearRegressionWithSGD]
```

Comment:
Why can't we do

```scala
abstract class StreamingLinearAlgorithm[
    M <: GeneralizedLinearModel,
    A <: GeneralizedLinearAlgorithm[M]] extends Logging with StreamingDecay[A] {
```
since `A` should have the concrete subtype